File: 11-ContainerWithMostWater/11-ContainerWithMostWater.cpp
// Last updated: 08/05/2026, 09:34:01
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left=0;
        int right= height.size()-1;
        int maxarea=0;
        int area=0;
        while(left<right){
            area= (right-left)* min(height[left], height[right]);
            if(height[left]<=height[right]){
                left+=1;
            }           
            else{
                right-=1;
            } 

            maxarea= max(maxarea, area);
        }

        return maxarea;
        
    }
};

File: 11-ContainerWithMostWater/11-ContainerWithMostWater.py
# Last updated: 08/05/2026, 09:34:05
class Solution:
    def maxArea(self, height: List[int]) -> int:
        

        left=0
        right=len(height)-1
        prod=0
    
        while left<right:

            value= min(height[left], height[right])*(right-left)

            if value>prod:
                prod= value

            if height[left]<height[right]:
                left+=1
            else:
                right-=1

        return prod